R/run.R

```diff
@@ -390,7 +390,15 @@ CmdStanRun$set("private", name = "run_generate_quantities_", value = .run_genera
     }
     procs$set_active_procs(procs$num_alive())
   }
-  if (procs$get_proc(id)$get_exit_status() == 0) {
+  successful_fit = FALSE
+  if (self$method() == "optimize") {
+    if (procs$proc_state(id = id) > 3) {
```

Comment:
This marks that either Optimization terminated with error or Optimization terminated normally was printed which means that optimization terminated "normally".